src/software/ai/hl/stp/tactic/tactic.h

```diff
@@ -10,6 +10,12 @@
 #include "software/ai/intent/intent.h"
 #include "software/world/world.h"
 
+#define COPY_CLASS(new_class, parent_class)                                              \
+    class new_class : public parent_class                                                \
+    {                                                                                    \
+        using parent_class::parent_class;                                                \
+    };
+
```

Comment:
```suggestion
#define COPY_CLASS(new_class, parent_class)                                              \
    class new_class : public parent_class                                                \
    {                                                                                    \
        using parent_class::parent_class;                                                \
                                                                                         \
        void accept(TacticVisitor &visitor) const                                        \
        {                                                                                \
            visitor.visit(*this);                                                        \
        }                                                                                \
    };

```
The tactic visitor needs to be overridden to have different motion constraints